src/hotspot/share/opto/loopPredicate.cpp

```diff
@@ -1350,8 +1358,16 @@ bool PhaseIdealLoop::loop_predication_impl(IdealLoopTree* loop) {
   CountedLoopNode *cl = nullptr;
   if (head->is_valid_counted_loop(T_INT)) {
     cl = head->as_CountedLoop();
-    // do nothing for iteration-splitted loops
-    if (!cl->is_normal_loop()) return false;
+    if (!cl->is_normal_loop()) {
+      // Do nothing for iteration-splitted loops
+      return false;
+    }
+    loop->compute_trip_count(this);
+    if (cl->trip_count() == 1) {
```

Comment:
I don't check for `has_exact_trip_count()` here since we could also have non-`ConNodes` with a type such that the iv phi is also a constant and we know that we will only iterate for at most one iteration.